### DISCUSSION TITLE
Use it without a SD card

### DISCUSSION THREAD

**Discussion by wdhzw** [jomjol/AI-on-the-edge-device#2497] [Q&A]
Just wondering can this product work without using a SD card, i.e. store the html files, reference images and roi coordinates in the esp-cam memory?

**Comment by caco3** [comment]
No, because the internal flash is only 2 MB and is almost full (used to 93%).

**Reply by wdhzw** [reply]
Thanks for the reply. But I just checked that esp32-cam has 4MB flash, is 4 MB enough for this?

**Reply by caco3** [reply]
hmm, but I guess due to the partitioning, we only can use 2 MB:
` Flash: [========= ]  93.3% (used 1815009 bytes from 1945600 bytes)`

And even if you would have enough memory available, the firmware is not made to serve the mentioned data from internal flash.

**Comment by Sillium** [comment]
Is there any theoretical way to omit the SD card at some point in the future? I had the second SD card failure within days. Any chance of using a network drive or attaching some kind of SSD via Micro USB or something similar?

**Reply by Slider0007** [reply]
With actual concept it's not so easy to completely get rid of SD card, but it would be possible to further reduce SD card r/w cycles. I've created a fork of this software for my own usage with a slight conceptual change where no sd card r/w occurs during regular processing as long as no debug or data logging is used. All data (models, reference images, ...) which are necessary for processing get loaded into RAM only once during process init.

Firstly, I would ensure that debug logging level is set to 'error' or 'warning' and as long you do not need local data logging, it's switched off.

Try to use only quality SD cards. (Reading endurance is important, because most of time SD card is only used for reading data, e.g. loading models from SD card every cycle)

**Reply by Sillium** [reply]
This fork with ramdisk usage sounds super interesting. Any chance to merge this feature into the main project? I reduced logging and retention to the minimum anyway to reduce strain on the SD card.

**Reply by caco3** [reply]
I am really wondering why your cards dies. I have 2 devices where I have an interval of 2 minutes, logging is set to info and data logging is enabled. nevertheless both cards work fine now since almost a year!

**Reply by Sillium** [reply]
Maybe those were not really dying. I used 2 cards from Intenso with 8 GB and both worked flunky. One worked, then stopped working but was working again after I reformatted and put the contents back on it. Then it stopped working again and I could not repair it again. The second one just did the same and I didn't want to repeat the steps every day, so I bought some other 8 GB cards in the meantime. Now I'm using a SanDisk Industrial since two days (interval 15 minutes). Maybe that one will work for a significantly longer time and the problems were caused by the cheap Intenso cards.

**Comment by friedpa** [comment]
I have invested in two SanDisk Extreme and they are working now since the beginning of this project for about two years (Intervall 3 minutes).
